src/javascript/app/pages/user/metatrader/metatrader.ui.js

```diff
@@ -380,6 +384,14 @@ const MetaTraderUI = (() => {
         $form.find('.mt5_type_box').click(selectAccountTypeUI);
     };
 
+    // restoreUI:
+    // On deposit response setCurrentAccount is called, which updates DOM
```

Comment:
what if instead of this, we check before updating DOM if you are on the expected action?